### ISSUE TITLE
Virtua Racing issue

### ISSUE THREAD

**Issue by mywebsiteworks2** [yoshisuga/MAME4iOS#349]
Using the MAME 0.234 core, Virtua Racing which should've been playable and working isn't working properly (and IIRC has the same issue the 0.139u1 had). Using iPhone 11

https://user-images.githubusercontent.com/49933793/132085885-b4f2e1f4-0421-4f90-82e3-b75be3c5d781.mp4

**Comment by ToddLa** [comment]
Sounds like a controller mapping issue, especially if it does the same thing in 139 too.

Summon the MAME Configure menu and look at how the inputs are mapped.

I will take a look too.

**Comment by ToddLa** [comment]
Virtua Racing gets input from a Paddle and Peddle, these should get mapped to the game controller triggers, will investigate

**Comment by mywebsiteworks2** [comment]
Don’t think this is have anything to do with control mapping (looks more like a emulation bug to me), but whatever

![Uploading FF1B99D3-0364-4F00-BA4F-F5679655650B.png…]()

**Comment by mywebsiteworks2** [comment]
Dont think it has anything to do with the controller mapping (looks more like an emulation bug to me), but whatever 

![IMG-20210905-WA0000](https://user-images.githubusercontent.com/49933793/132113172-fec14758-a2f9-40d1-bbd1-1faa1277c836.jpg)

**Comment by mywebsiteworks2** [comment]
In fact, here’s a bit of the attract mode footage

https://streamable.com/u5um68

(Sorry for the external link. I’m not near my computer right now, so I have no means to properly compress the videos).

**Comment by ToddLa** [comment]
> Dont think it has anything to do with the controller mapping (looks more like an emulation bug to me), but whatever 
> 
> 
> 
> ![IMG-20210905-WA0000](https://user-images.githubusercontent.com/49933793/132113172-fec14758-a2f9-40d1-bbd1-1faa1277c836.jpg)
> 
> 

Yup those mappings look ok, on screen controls should work (might want to switch to analog in Settings > Game Input) but the INC and DEC will function.

**Comment by ToddLa** [comment]
> In fact, here’s a bit of the attract mode footage
> 
> 
> 
> https://streamable.com/u5um68
> 
> 
> 
> (Sorry for the external link. I’m not near my computer right now, so I have no means to properly compress the videos).

Wow, that is messed up! I will check it out!

**Comment by ToddLa** [comment]
The same thing happens in baseline MAME 0235

this is NOT a MAME4iOS issue, please close and report to MAME

**Comment by mywebsiteworks2** [comment]
Wait, I played this on mainline MAME 0.234 and it was fine. I’ll give the direct capture from my laptop soon.

**Comment by ToddLa** [comment]
Hmmm I just built 0235 and ran, and had the same issues you had (spinning to the left in attach mode etc)

Maybe it is a problem when built for arm64 (I built on a M1) so same problem for iOS and M1.

I will fire up a build on my old MBP and go see a double feature....

**Comment by ToddLa** [comment]
Sorry @mywebsiteworks2 for the back and forth, thanks for isolating this....

**Comment by mywebsiteworks2** [comment]
Not sure if the problem's caused by the ARM64 build
This is what it should look like on normal X86 MAME 0.235 (and 0.234 as well)

https://user-images.githubusercontent.com/49933793/132303429-a8ab4765-d68c-4916-bf41-b9ebc6898eec.mp4

**Comment by ToddLa** [comment]
My x64 build of MAME runs VR just fine.  So it is a problem with the arm64 build, maybe DRC.

The input is a little strange, but that will be a issue later when we get the arm64 to build right.

**Comment by ToddLa** [comment]
Need to investigate flags on arm build.

**Comment by ToddLa** [comment]
Like other DRC game (NFL Blitz, ...) the arm64 c-drc in MAME does not work.

**Comment by avv3lenato** [comment]
Same problem on the latest MAME arm64 build for Mac running on M2
